### PR TITLE
Server with Thread Pool Updated

### PR DESCRIPTION
**Pull request by hooohu**

### CODE REVIEW

**Comment by phil-lopreiato** [comment]
@atryus will do the full review for this PR, but I saw a few style guide issues (make sure you have consistent 8-space indents and have return types for functions on their own lines). Also, make sure you're thinking about writing unit tests for the code (necessary before it'll get merged to master)

File: server/Makefile [hunk 1]
```diff
@@ -15,8 +16,8 @@ all: $(OBJS)
 
 rebuild: clean all
 
-# server-kv: ht-wrapper server-kv.c
-#	$(CC) $(CFLAGS) -o $@ $^
+$(OBJS): ht-wrapper
+	$(CC) $(CFLAGS) -o $(OBJS) $(SRCS)
```
Comment (amygara) [inline server/Makefile]: @nks5295 needs to link in `gwkv_ht_wrapper.o` _ignore for now_

File: server/server-kv.c [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+struct pool_list {
+	int fd;
+	struct pool_list *next;
+} *list_head = NULL, *list_tail = NULL;
+
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+pthread_t threads[MAX_CONCURRENCY];
+
+void parse_message(char *message, int len, struct operation *operation) {
```
Comment (amygara) [inline server/server-kv.c]: style guide plz

File: server/server-kv.c [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        return;
+}
+
+void process_operation(struct operation *operation, char *message, int *len) {
```
Comment (amygara) [inline server/server-kv.c]: style guide plz

File: server/server-kv.c [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        return;
+}
+
+/* Handle a request from a client */
+void *handle_request(void *ptr){
```
Comment (amygara) [inline server/server-kv.c]: style guide plz

File: server/server-kv.c [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        int bytes_read, bytes_write;
```
Comment (amygara) [inline server/server-kv.c:32]: spacing

File: server/server-kv.c [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        char message[256];
```
Comment (amygara) [inline server/server-kv.c:33]: spacing

File: server/server-kv.c [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                //
+
+                close(clientfd);
+		free(node);
+	}
+}
+
+
+/* Main server logic */
+void server_main(int sockfd, char* thread_number) {
```
Comment (amygara) [inline server/server-kv.c]: style guide plz

File: server/server-kv.c [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+		else {
+			list_tail->next = node;
+		}
+		list_tail = node;
+		pthread_cond_signal(&cond);
+		pthread_mutex_unlock(&mutex);
+        }
+}
+
+int main(int argc, char ** argv)
```
Comment (amygara) [inline server/server-kv.c]: style guide plz

File: server/server-kv.c [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+	int i, tnum;
+	struct pool_list *node;
+
+        tnum = atoi(thread_number);
+	for(i = 0; i < tnum; i++) {
+		pthread_create(&threads[i], NULL, handle_request, NULL);
+	}
+
+        while(1) {
+                struct sockaddr_storage client_addr;
```
Comment (amygara) [inline server/server-kv.c:82]: 104-110, spacing

File: server/server-kv.c [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	}
+}
+
+
+/* Main server logic */
+void server_main(int sockfd, char* thread_number) {
+	int i, tnum;
+	struct pool_list *node;
+
+        tnum = atoi(thread_number);
```
Comment (amygara) [inline server/server-kv.c:76]: spacing

File: server/server-kv.c [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+
+	while (1) {
+		pthread_mutex_lock(&mutex);
+		while (list_head == NULL) {
+			pthread_cond_wait(&cond, &mutex);
+		}
+		node = list_head;
+		list_head = node->next;
+		pthread_mutex_unlock(&mutex);
+                clientfd = node->fd;
```
Comment (amygara) [inline server/server-kv.c:46]: spacing

File: server/server-kv.c [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+	        
+                //for test	
+                bytes_read = read(clientfd, message, 256);
```
Comment (amygara) [inline server/server-kv.c:49]: spacing

File: server/server-kv.c [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+		}
+		else {
+			parse_message(message, bytes_read, &operation);
+			process_operation(&operation, message, &bytes_write);
+			write(clientfd, message, bytes_write);
+                        printf("thread: %d, received:%s\n", pthread_self(), message);
+		}
+                //
+
+                close(clientfd);
```
Comment (amygara) [inline server/server-kv.c:64]: spacing

**Pull request merged**